service/pkg/server/services.go

```diff
@@ -194,13 +194,14 @@ func startServices(ctx context.Context, params startServicesParams) (func(), err
 				svcLogger = svcLogger.With("version", svc.GetVersion())
 			}
 
-			// Check if the service supports and needs a cache
-			var cacheClient *cache.Cache
-			if cacheSvc, ok := svc.(serviceregistry.CacheSupportedService); ok {
-				cacheClient, err = cacheManager.NewCache(ns, svcLogger, *cacheSvc.CacheOptions())
+			// Function to create a cache given cache options
+			var createCacheClient func(cache.Options) (*cache.Cache, error) = func(options cache.Options) (*cache.Cache, error) {
+				slog.Info("creating cache client for", slog.String("namespace", ns), slog.String("service", svc.GetServiceDesc().ServiceName))
+				cacheClient, err := cacheManager.NewCache(fmt.Sprintf("%s-%s", ns, svc.GetServiceDesc().ServiceName), svcLogger, options)
```

Comment:
The namespace is already unique so we shouldn't need the ServiceName.

Reply:
what about services under policy?

Reply:
resolved: one cache per ns